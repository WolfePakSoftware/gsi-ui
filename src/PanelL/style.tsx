import styled from 'styled-components';

export const PanelContainer = styled.div`
  border: 1px solid #d4d4d5;
  border-top: none;
`;

export const styleHeader = {
  borderBottomLeftRadius: 0,
  borderBottomRightRadius: 0,
  padding: 5,
  margin: 0
};

export const PanelBody = styled.div`
  padding: 10px;
`;