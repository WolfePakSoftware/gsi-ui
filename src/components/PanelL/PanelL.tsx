import { Header } from 'semantic-ui-react';
import { PanelBody, PanelContainer, styleHeader } from './style';

interface PanelProps {
  header: string;
  children?: any;
}

import React, { Component } from 'react';

export class PanelL extends Component<PanelProps> {
  constructor(props:PanelProps) {
    super(props);
  }

  render() {
    return (
      <React.Fragment>
        <Header as="h2" textAlign="center" block size="tiny" style={styleHeader}>
          {this.props.header}
        </Header>
        <PanelContainer>
          <PanelBody>{this.props.children}</PanelBody>
        </PanelContainer>
      </React.Fragment>
    );
  }
}