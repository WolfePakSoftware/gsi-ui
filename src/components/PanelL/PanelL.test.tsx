import React from "react";
import ReactDOM from "react-dom";
import {PanelL} from "./PanelL";
import renderer from 'react-test-renderer';
import 'jest-styled-components';

it("renders without crashing Progress Label", () => {
  const div = document.createElement("div");
  const header = 'Hola mundo';
  const child = (<div><h1>Soy yo</h1></div>);
  ReactDOM.render(<PanelL header={header} children={child}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('render correctly PanelL', () => {
  const header = 'Hola mundo';
  const child = (<div><h1>Soy yo</h1></div>);
  const TextInputComponent = renderer.create(<PanelL header={header} children={child}/>).toJSON();
  expect(TextInputComponent).toMatchSnapshot();
});