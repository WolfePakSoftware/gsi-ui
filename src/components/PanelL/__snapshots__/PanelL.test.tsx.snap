// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly PanelL 1`] = `
Array [
  <h2
    className="ui tiny block center aligned header"
    style={
      Object {
        "borderBottomLeftRadius": 0,
        "borderBottomRightRadius": 0,
        "margin": 0,
        "padding": 5,
      }
    }
  >
    Hola mundo
  </h2>,
  .c0 {
  border: 1px solid #d4d4d5;
  border-top: none;
}

.c1 {
  padding: 10px;
}

<div
    className="c0"
  >
    <div
      className="c1"
    >
      <div>
        <h1>
          Soy yo
        </h1>
      </div>
    </div>
  </div>,
]
`;
